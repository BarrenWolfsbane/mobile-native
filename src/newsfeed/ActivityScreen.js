import React, { Component } from 'react';

import {
  View,
  Text
} from 'react-native';
import { observer } from 'mobx-react/native'
import FastImage from 'react-native-fast-image';

import NewsfeedStore from "./NewsfeedStore";
import { getSingle } from './NewsfeedService';
import { CommonStyle as CS } from '../styles/Common';
import CommentList from '../comments/CommentList';
import Activity from '../newsfeed/activity/Activity';
import ActivityModel from '../newsfeed/ActivityModel';
import { ComponentsStyle } from '../styles/Components';
import SingleEntityStore from '../common/stores/SingleEntityStore';
import CenteredLoading from '../common/components/CenteredLoading';
import commentsStoreProvider from '../comments/CommentsStoreProvider';
import logService from '../common/services/log.service';
import i18n from '../common/services/i18n.service';

/**
 * Activity screen
 */
@observer
export default class ActivityScreen extends Component {

  static navigationOptions = ({ navigation }) => {
    return {
      title: navigation.getParam('entity', {ownerObj:{name:''}}).ownerObj.name
    };
  };

  entityStore = new SingleEntityStore();

  state = {
    error: null
  };

  /**
   * Constructor
   * @param {object} props
   */
  constructor(props) {
    super(props);

    const params = props.navigation.state.params;

    this.comments = commentsStoreProvider.get();

    if (params.entity && (params.entity.guid || params.entity.entity_guid)) {
      this.entityStore.setEntity(ActivityModel.checkOrCreate(params.entity));

      const entity = this.entityStore.entity;

      if (entity._list && entity._list.metadataServie) {
        entity._list.metadataServie.pushSource('single');
      }
    }
  }

  /**
   * Component did mount
   */
  async componentDidMount() {
    const params = this.props.navigation.state.params;

    if (!this.entityStore.entity || params.hydrate) {
      try {
        const resp = await getSingle(params.guid || params.entity.guid || params.entity.entity_guid);

        // if it has a list asigned we set it to the new entity
        if (this.entityStore.entity) {
          this.entityStore.entity.update(resp.activity);
        } else {
          this.entityStore.setEntity(ActivityModel.checkOrCreate(resp.activity));
        }
      } catch (e) {
        this.setState({error: true});
        logService.exception('[ActivityScreen]',e);
        //console.error('Cannot hydrate activity', e);
      }
    }
  }

  /**
   * Component will unmount
   */
  componentWillUnmount() {
    const entity = this.entityStore.entity;

    if (entity._list && entity._list.metadataServie) {
      entity._list.metadataServie.popSource();
    }
  }

  /**
   * Get header
   */
  getHeader() {
    return this.entityStore.entity ?
      <Activity
        ref={o => this.activity = o}
        entity={ this.entityStore.entity }
        navigation={ this.props.navigation }
        autoHeight={false}
      /> : null;
  }

  /**
   * On comment input focus
   */
  onFocus = () => {
    this.activity.pauseVideo();
  }

  /**
   * Render
   */
  render() {
    if (!this.entityStore.entity && !this.state.error) return <CenteredLoading />;
    return (
      <View style={[CS.flexContainer, CS.backgroundWhite]}>
        {
          !this.state.error ?
            <CommentList
              header={this.getHeader()}
              entity={this.entityStore.entity}
              store={this.comments}
              navigation={this.props.navigation}
              onInputFocus={this.onFocus}
            />
          :
            <View style={CS.flexColumnCentered}>
              <FastImage
                resizeMode={FastImage.resizeMode.contain}
                style={ComponentsStyle.logo}
                source={require('../assets/logos/logo.png')}
              />
              <Text style={[CS.fontL, CS.colorDanger]}>{i18n.t('activity.error')}</Text>
              <Text style={[CS.fontM]}>{i18n.t('activity.tryAgain')}</Text>
            </View>
        }
      </View>
    );
  }
}