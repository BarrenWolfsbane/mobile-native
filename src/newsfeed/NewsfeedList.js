import React, { Component } from 'react';
import { FlatList, StyleSheet, View } from 'react-native';
import { observer } from 'mobx-react/native'

import Activity from './activity/Activity';
import Poster from './Poster';

/**
 * News feed list component
 */
@observer
export default class NewsfeedList extends Component {

  /**
   * Render component
   */
  render() {
    return (
      <FlatList
        ListHeaderComponent={<Poster/>}
        data={this.props.newsfeed.entities.slice()}
        renderItem={this.renderActivity}
        keyExtractor={item => item.guid}
        onRefresh={this.refresh}
        refreshing={this.props.newsfeed.refreshing}
        onEndReached={this.loadFeed}
        onEndThreshold={0}
        style={styles.listView}
      />
    );
  }

  /**
   * Load feed data
   */
  loadFeed = () => {
    this.props.newsfeed.loadFeed(this.props.guid);
  }

  /**
   * Refresh feed data
   */
  refresh = () => {
    this.props.newsfeed.refresh(this.props.guid)
  }

  /**
   * Render row
   */
  renderActivity = (row) => {
    const entity = row.item;
    return (
      <View>
        <Activity entity={entity} navigation={this.props.navigation} />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  listView: {
    //paddingTop: 20,
    backgroundColor: '#FFF',
    flex: 1,
  }
});