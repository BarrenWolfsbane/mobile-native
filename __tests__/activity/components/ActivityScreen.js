import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import {
  FlatList,
  KeyboardAvoidingView
} from 'react-native';

import ActivityScreen from '../../../src/newsfeed/ActivityScreen';
// import SingleEntityStore from '../../../src/common/stores/SingleEntityStore';
import RichEmbedStore from '../../../src/common/stores/RichEmbedStore';
import commentsStoreProvider from '../../../src/comments/CommentsStoreProvider';

import { commentsServiceFaker } from '../../../__mocks__/fake/CommentsFaker';
import { activitiesServiceFaker } from '../../../__mocks__/fake/ActivitiesFaker';
import CommentList from '../../../src/comments/CommentList';
jest.mock('../../../src/newsfeed/NewsfeedService');
import { getSingle } from '../../../src/newsfeed/NewsfeedService';
import entitiesService from '../../../src/common/services/entities.service';

jest.mock('../../../src/common/BaseModel');
jest.mock('../../../src/newsfeed/activity/Activity', () => 'Activity');
jest.mock('../../../src/comments/CommentList', () => 'CommentList');
jest.mock('../../../src/common/components/CenteredLoading', () => 'CenteredLoading');
jest.mock('../../../src/comments/CommentsStore');
jest.mock('../../../src/comments/CommentsStoreProvider');
jest.mock('../../../src/common/services/entities.service');


describe('Activity screen component', () => {

  let user, comments, entity, screen, navigation;
  beforeEach(() => {
    let mockResponse = commentsServiceFaker().load(5);
    commentsStoreProvider.get.mockReturnValue({
      comments: mockResponse.comments,
      embed: new RichEmbedStore(),
      loadNext: mockResponse['load-next'],
      loadPrevious: mockResponse['load-previous'],
      setText: () => {},
      post: () => {},
      loadComments: () => {
        return mockResponse.comments;
      },
      attachment:{}
    });
  });

  it('renders correctly with an entity as param', async () => {
    navigation = {
      push: jest.fn(),
      state: {
        routeName: 'some',
        params: {entity: activitiesServiceFaker().load(1).activities[0]}
      }
    };

    entitiesService.single.mockResolvedValue(navigation.state.params.entity);

    screen = shallow(
      <ActivityScreen navigation={navigation}/>
    );

    // shoul show loading
    expect(screen).toMatchSnapshot();

    // unmount
    await screen.instance().componentDidMount();

    jest.runAllTicks();

    // await is important here!
    await screen.update();

    // should show the activity
    expect(screen).toMatchSnapshot();
    // should have a comment list component
    expect(screen.find(CommentList)).toHaveLength(1);
  });
});
