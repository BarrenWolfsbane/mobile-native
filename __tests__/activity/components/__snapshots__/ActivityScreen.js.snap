// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Activity screen component renders correctly with an entity as param 1`] = `
<ForwardRef(View)
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "backgroundColor": "white",
      },
    ]
  }
>
  <CommentList
    entity={
      ActivityModel {
        "attachment_guid": false,
        "blurb": false,
        "container_guid": "activityguid0",
        "custom_data": false,
        "custom_type": false,
        "description": "Congratulations! ",
        "edited": "",
        "getThumbSource": [Function],
        "guid": "activityguid0",
        "is_visible": false,
        "mature": false,
        "mature_visibility": false,
        "message": "Message",
        "ownerObj": UserModel {
          "getAvatarSource": [Function],
          "guid": "824853017709780997",
          "subtype": false,
          "time_created": "1522036284",
          "type": "user",
        },
        "owner_guid": "824853017709780997",
        "parent_guid": "838106762591510528",
        "paywall": undefined,
        "perma_url": false,
        "rowKey": "something0",
        "thumbnail_src": false,
        "thumbs:down:count": undefined,
        "thumbs:down:user_guids": undefined,
        "thumbs:up:count": undefined,
        "thumbs:up:user_guids": undefined,
        "title": "TITLE",
        "type": "activity",
        "wire_totals": Object {
          "tokens": 1000000000000000000,
        },
        Symbol(mobx did run lazy initializers): true,
        Symbol(mobx administration): ObservableObjectAdministration$$1 {
          "defaultEnhancer": [Function],
          "keysAtom": Atom$$1 {
            "diffValue": 0,
            "isBeingObserved": false,
            "isPendingUnobservation": false,
            "lastAccessedBy": 0,
            "lowestObserverState": 2,
            "name": "ActivityModel@7.keys",
            "observers": Set {},
          },
          "name": "ActivityModel@7",
          "target": [Circular],
          "values": Map {
            "votedUp" => false,
            "votedDown" => false,
            "mature_visibility" => false,
            "is_visible" => false,
            "thumbs:down:count" => undefined,
            "thumbs:up:count" => undefined,
            "paywall" => undefined,
            "mature" => false,
            "edited" => "",
            "thumbs:down:user_guids" => undefined,
            "thumbs:up:user_guids" => undefined,
          },
        },
      }
    }
    header={
      <Activity
        autoHeight={true}
        entity={
          ActivityModel {
            "attachment_guid": false,
            "blurb": false,
            "container_guid": "activityguid0",
            "custom_data": false,
            "custom_type": false,
            "description": "Congratulations! ",
            "edited": "",
            "getThumbSource": [Function],
            "guid": "activityguid0",
            "is_visible": false,
            "mature": false,
            "mature_visibility": false,
            "message": "Message",
            "ownerObj": UserModel {
              "getAvatarSource": [Function],
              "guid": "824853017709780997",
              "subtype": false,
              "time_created": "1522036284",
              "type": "user",
            },
            "owner_guid": "824853017709780997",
            "parent_guid": "838106762591510528",
            "paywall": undefined,
            "perma_url": false,
            "rowKey": "something0",
            "thumbnail_src": false,
            "thumbs:down:count": undefined,
            "thumbs:down:user_guids": undefined,
            "thumbs:up:count": undefined,
            "thumbs:up:user_guids": undefined,
            "title": "TITLE",
            "type": "activity",
            "wire_totals": Object {
              "tokens": 1000000000000000000,
            },
            Symbol(mobx did run lazy initializers): true,
            Symbol(mobx administration): ObservableObjectAdministration$$1 {
              "defaultEnhancer": [Function],
              "keysAtom": Atom$$1 {
                "diffValue": 0,
                "isBeingObserved": false,
                "isPendingUnobservation": false,
                "lastAccessedBy": 0,
                "lowestObserverState": 2,
                "name": "ActivityModel@7.keys",
                "observers": Set {},
              },
              "name": "ActivityModel@7",
              "target": [Circular],
              "values": Map {
                "votedUp" => false,
                "votedDown" => false,
                "mature_visibility" => false,
                "is_visible" => false,
                "thumbs:down:count" => undefined,
                "thumbs:up:count" => undefined,
                "paywall" => undefined,
                "mature" => false,
                "edited" => "",
                "thumbs:down:user_guids" => undefined,
                "thumbs:up:user_guids" => undefined,
              },
            },
          }
        }
        navigation={
          Object {
            "push": [MockFunction],
            "state": Object {
              "params": Object {
                "entity": Object {
                  "attachment_guid": false,
                  "blurb": false,
                  "container_guid": "activityguid0",
                  "custom_data": false,
                  "custom_type": false,
                  "description": "Congratulations! ",
                  "edited": "",
                  "getThumbSource": [Function],
                  "guid": "activityguid0",
                  "mature": false,
                  "message": "Message",
                  "ownerObj": Object {
                    "getAvatarSource": [Function],
                    "guid": "824853017709780997",
                    "subtype": false,
                    "time_created": "1522036284",
                    "type": "user",
                  },
                  "owner_guid": "824853017709780997",
                  "parent_guid": "838106762591510528",
                  "perma_url": false,
                  "rowKey": "something0",
                  "thumbnail_src": false,
                  "title": "TITLE",
                  "type": "activity",
                  "wire_totals": Object {
                    "tokens": 1000000000000000000,
                  },
                },
              },
              "routeName": "some",
            },
          }
        }
        newsfeed={
          NewsfeedStore {
            "boosts": Array [],
            "filter": "subscribed",
            "loadingBoost": true,
            "service": NewsfeedService {
              "_getFeed": [MockFunction],
              "getBoosts": [MockFunction],
              "getFeed": [MockFunction],
              "getFeedChannel": [MockFunction],
              "getFeedSuggested": [MockFunction],
            },
            "stores": Object {
              "boostfeed": Object {
                "list": OffsetFeedListStore {
                  "entities": Array [],
                  Symbol(mobx administration): ObservableObjectAdministration$$1 {
                    "defaultEnhancer": [Function],
                    "keysAtom": Atom$$1 {
                      "diffValue": 0,
                      "isBeingObserved": false,
                      "isPendingUnobservation": false,
                      "lastAccessedBy": 0,
                      "lowestObserverState": 2,
                      "name": "OffsetFeedListStore@6.keys",
                      "observers": Set {},
                    },
                    "name": "OffsetFeedListStore@6",
                    "target": [Circular],
                    "values": Map {
                      "entities" => Array [],
                    },
                  },
                },
                "loading": false,
              },
              "subscribed": Object {
                "list": OffsetFeedListStore {
                  "entities": Array [
                    ActivityModel {
                      "attachment_guid": false,
                      "blurb": false,
                      "container_guid": "activityguid0",
                      "custom_data": false,
                      "custom_type": false,
                      "description": "Congratulations! ",
                      "edited": "",
                      "getThumbSource": [Function],
                      "guid": "activityguid0",
                      "is_visible": false,
                      "mature": false,
                      "mature_visibility": false,
                      "message": "Message",
                      "ownerObj": UserModel {
                        "getAvatarSource": [Function],
                        "guid": "824853017709780997",
                        "subtype": false,
                        "time_created": "1522036284",
                        "type": "user",
                      },
                      "owner_guid": "824853017709780997",
                      "parent_guid": "838106762591510528",
                      "paywall": undefined,
                      "perma_url": false,
                      "rowKey": "something0",
                      "thumbnail_src": false,
                      "thumbs:down:count": undefined,
                      "thumbs:down:user_guids": undefined,
                      "thumbs:up:count": undefined,
                      "thumbs:up:user_guids": undefined,
                      "title": "TITLE",
                      "type": "activity",
                      "wire_totals": Object {
                        "tokens": 1000000000000000000,
                      },
                      Symbol(mobx did run lazy initializers): true,
                      Symbol(mobx administration): ObservableObjectAdministration$$1 {
                        "defaultEnhancer": [Function],
                        "keysAtom": Atom$$1 {
                          "diffValue": 0,
                          "isBeingObserved": false,
                          "isPendingUnobservation": false,
                          "lastAccessedBy": 0,
                          "lowestObserverState": 2,
                          "name": "ActivityModel@7.keys",
                          "observers": Set {},
                        },
                        "name": "ActivityModel@7",
                        "target": [Circular],
                        "values": Map {
                          "votedUp" => false,
                          "votedDown" => false,
                          "mature_visibility" => false,
                          "is_visible" => false,
                          "thumbs:down:count" => undefined,
                          "thumbs:up:count" => undefined,
                          "paywall" => undefined,
                          "mature" => false,
                          "edited" => "",
                          "thumbs:down:user_guids" => undefined,
                          "thumbs:up:user_guids" => undefined,
                        },
                      },
                    },
                  ],
                  Symbol(mobx administration): ObservableObjectAdministration$$1 {
                    "defaultEnhancer": [Function],
                    "keysAtom": Atom$$1 {
                      "diffValue": 0,
                      "isBeingObserved": false,
                      "isPendingUnobservation": false,
                      "lastAccessedBy": 0,
                      "lowestObserverState": 2,
                      "name": "OffsetFeedListStore@4.keys",
                      "observers": Set {},
                    },
                    "name": "OffsetFeedListStore@4",
                    "target": [Circular],
                    "values": Map {
                      "entities" => Array [
                        ActivityModel {
                          "attachment_guid": false,
                          "blurb": false,
                          "container_guid": "activityguid0",
                          "custom_data": false,
                          "custom_type": false,
                          "description": "Congratulations! ",
                          "edited": "",
                          "getThumbSource": [Function],
                          "guid": "activityguid0",
                          "is_visible": false,
                          "mature": false,
                          "mature_visibility": false,
                          "message": "Message",
                          "ownerObj": UserModel {
                            "getAvatarSource": [Function],
                            "guid": "824853017709780997",
                            "subtype": false,
                            "time_created": "1522036284",
                            "type": "user",
                          },
                          "owner_guid": "824853017709780997",
                          "parent_guid": "838106762591510528",
                          "paywall": undefined,
                          "perma_url": false,
                          "rowKey": "something0",
                          "thumbnail_src": false,
                          "thumbs:down:count": undefined,
                          "thumbs:down:user_guids": undefined,
                          "thumbs:up:count": undefined,
                          "thumbs:up:user_guids": undefined,
                          "title": "TITLE",
                          "type": "activity",
                          "wire_totals": Object {
                            "tokens": 1000000000000000000,
                          },
                          Symbol(mobx did run lazy initializers): true,
                          Symbol(mobx administration): ObservableObjectAdministration$$1 {
                            "defaultEnhancer": [Function],
                            "keysAtom": Atom$$1 {
                              "diffValue": 0,
                              "isBeingObserved": false,
                              "isPendingUnobservation": false,
                              "lastAccessedBy": 0,
                              "lowestObserverState": 2,
                              "name": "ActivityModel@7.keys",
                              "observers": Set {},
                            },
                            "name": "ActivityModel@7",
                            "target": [Circular],
                            "values": Map {
                              "votedUp" => false,
                              "votedDown" => false,
                              "mature_visibility" => false,
                              "is_visible" => false,
                              "thumbs:down:count" => undefined,
                              "thumbs:up:count" => undefined,
                              "paywall" => undefined,
                              "mature" => false,
                              "edited" => "",
                              "thumbs:down:user_guids" => undefined,
                              "thumbs:up:user_guids" => undefined,
                            },
                          },
                        },
                      ],
                    },
                  },
                },
                "loading": false,
              },
              "suggested": Object {
                "list": OffsetFeedListStore {
                  "entities": Array [],
                  Symbol(mobx administration): ObservableObjectAdministration$$1 {
                    "defaultEnhancer": [Function],
                    "keysAtom": Atom$$1 {
                      "diffValue": 0,
                      "isBeingObserved": false,
                      "isPendingUnobservation": false,
                      "lastAccessedBy": 0,
                      "lowestObserverState": 2,
                      "name": "OffsetFeedListStore@5.keys",
                      "observers": Set {},
                    },
                    "name": "OffsetFeedListStore@5",
                    "target": [Circular],
                    "values": Map {
                      "entities" => Array [],
                    },
                  },
                },
                "loading": false,
              },
            },
            "viewed": Array [],
            Symbol(mobx did run lazy initializers): true,
            Symbol(mobx administration): ObservableObjectAdministration$$1 {
              "defaultEnhancer": [Function],
              "keysAtom": Atom$$1 {
                "diffValue": 0,
                "isBeingObserved": false,
                "isPendingUnobservation": false,
                "lastAccessedBy": 0,
                "lowestObserverState": 2,
                "name": "NewsfeedStore@11.keys",
                "observers": Set {},
              },
              "name": "NewsfeedStore@11",
              "target": [Circular],
              "values": Map {
                "filter" => "subscribed",
                "boosts" => Array [],
              },
            },
          }
        }
      />
    }
    navigation={
      Object {
        "push": [MockFunction],
        "state": Object {
          "params": Object {
            "entity": Object {
              "attachment_guid": false,
              "blurb": false,
              "container_guid": "activityguid0",
              "custom_data": false,
              "custom_type": false,
              "description": "Congratulations! ",
              "edited": "",
              "getThumbSource": [Function],
              "guid": "activityguid0",
              "mature": false,
              "message": "Message",
              "ownerObj": Object {
                "getAvatarSource": [Function],
                "guid": "824853017709780997",
                "subtype": false,
                "time_created": "1522036284",
                "type": "user",
              },
              "owner_guid": "824853017709780997",
              "parent_guid": "838106762591510528",
              "perma_url": false,
              "rowKey": "something0",
              "thumbnail_src": false,
              "title": "TITLE",
              "type": "activity",
              "wire_totals": Object {
                "tokens": 1000000000000000000,
              },
            },
          },
          "routeName": "some",
        },
      }
    }
    onInputFocus={[Function]}
    store={
      Object {
        "attachment": Object {},
        "comments": Array [
          Object {
            "attachment_guid": false,
            "blurb": false,
            "container_guid": "guid0",
            "custom_data": false,
            "custom_type": false,
            "description": "Congratulations! ",
            "edited": "",
            "guid": "guid0",
            "mature": false,
            "ownerObj": Object {
              "guid": "824853017709780997",
              "subtype": false,
              "time_created": "1522036284",
              "type": "user",
            },
            "owner_guid": "824853017709780997",
            "parent_guid": "838106762591510528",
            "perma_url": false,
            "thumbnail_src": false,
            "type": "comment",
          },
          Object {
            "attachment_guid": false,
            "blurb": false,
            "container_guid": "guid1",
            "custom_data": false,
            "custom_type": false,
            "description": "Congratulations! ",
            "edited": "",
            "guid": "guid1",
            "mature": false,
            "ownerObj": Object {
              "guid": "824853017709780997",
              "subtype": false,
              "time_created": "1522036284",
              "type": "user",
            },
            "owner_guid": "824853017709780997",
            "parent_guid": "838106762591510528",
            "perma_url": false,
            "thumbnail_src": false,
            "type": "comment",
          },
          Object {
            "attachment_guid": false,
            "blurb": false,
            "container_guid": "guid2",
            "custom_data": false,
            "custom_type": false,
            "description": "Congratulations! ",
            "edited": "",
            "guid": "guid2",
            "mature": false,
            "ownerObj": Object {
              "guid": "824853017709780997",
              "subtype": false,
              "time_created": "1522036284",
              "type": "user",
            },
            "owner_guid": "824853017709780997",
            "parent_guid": "838106762591510528",
            "perma_url": false,
            "thumbnail_src": false,
            "type": "comment",
          },
          Object {
            "attachment_guid": false,
            "blurb": false,
            "container_guid": "guid3",
            "custom_data": false,
            "custom_type": false,
            "description": "Congratulations! ",
            "edited": "",
            "guid": "guid3",
            "mature": false,
            "ownerObj": Object {
              "guid": "824853017709780997",
              "subtype": false,
              "time_created": "1522036284",
              "type": "user",
            },
            "owner_guid": "824853017709780997",
            "parent_guid": "838106762591510528",
            "perma_url": false,
            "thumbnail_src": false,
            "type": "comment",
          },
          Object {
            "attachment_guid": false,
            "blurb": false,
            "container_guid": "guid4",
            "custom_data": false,
            "custom_type": false,
            "description": "Congratulations! ",
            "edited": "",
            "guid": "guid4",
            "mature": false,
            "ownerObj": Object {
              "guid": "824853017709780997",
              "subtype": false,
              "time_created": "1522036284",
              "type": "user",
            },
            "owner_guid": "824853017709780997",
            "parent_guid": "838106762591510528",
            "perma_url": false,
            "thumbnail_src": false,
            "type": "comment",
          },
        ],
        "embed": RichEmbedStore {
          "_richEmbedFetchTimer": null,
          "clearRichEmbedAction": [Function],
          "richEmbedUrl": "",
          "setRichEmbedPromise": null,
        },
        "loadComments": [Function],
        "loadNext": "aaaaaa",
        "loadPrevious": "aaaaaa",
        "post": [Function],
        "setText": [Function],
      }
    }
  />
</ForwardRef(View)>
`;

exports[`Activity screen component should show loader until it loads the activity 1`] = `<CenteredLoading />`;

exports[`Activity screen component should show loader until it loads the activity 2`] = `
<ForwardRef(View)
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "backgroundColor": "white",
      },
    ]
  }
>
  <CommentList
    entity={
      ActivityModel {
        "edited": undefined,
        "is_visible": false,
        "mature": undefined,
        "mature_visibility": false,
        "paywall": undefined,
        "thumbs:down:count": undefined,
        "thumbs:down:user_guids": undefined,
        "thumbs:up:count": undefined,
        "thumbs:up:user_guids": undefined,
        Symbol(mobx did run lazy initializers): true,
        Symbol(mobx administration): ObservableObjectAdministration$$1 {
          "defaultEnhancer": [Function],
          "keysAtom": Atom$$1 {
            "diffValue": 0,
            "isBeingObserved": false,
            "isPendingUnobservation": false,
            "lastAccessedBy": 0,
            "lowestObserverState": 2,
            "name": "ActivityModel@18.keys",
            "observers": Set {},
          },
          "name": "ActivityModel@18",
          "target": [Circular],
          "values": Map {
            "votedUp" => false,
            "votedDown" => false,
            "mature_visibility" => false,
            "is_visible" => false,
            "thumbs:down:count" => undefined,
            "thumbs:up:count" => undefined,
            "paywall" => undefined,
            "mature" => undefined,
            "edited" => undefined,
            "thumbs:down:user_guids" => undefined,
            "thumbs:up:user_guids" => undefined,
          },
        },
      }
    }
    header={
      <Activity
        autoHeight={true}
        entity={
          ActivityModel {
            "edited": undefined,
            "is_visible": false,
            "mature": undefined,
            "mature_visibility": false,
            "paywall": undefined,
            "thumbs:down:count": undefined,
            "thumbs:down:user_guids": undefined,
            "thumbs:up:count": undefined,
            "thumbs:up:user_guids": undefined,
            Symbol(mobx did run lazy initializers): true,
            Symbol(mobx administration): ObservableObjectAdministration$$1 {
              "defaultEnhancer": [Function],
              "keysAtom": Atom$$1 {
                "diffValue": 0,
                "isBeingObserved": false,
                "isPendingUnobservation": false,
                "lastAccessedBy": 0,
                "lowestObserverState": 2,
                "name": "ActivityModel@18.keys",
                "observers": Set {},
              },
              "name": "ActivityModel@18",
              "target": [Circular],
              "values": Map {
                "votedUp" => false,
                "votedDown" => false,
                "mature_visibility" => false,
                "is_visible" => false,
                "thumbs:down:count" => undefined,
                "thumbs:up:count" => undefined,
                "paywall" => undefined,
                "mature" => undefined,
                "edited" => undefined,
                "thumbs:down:user_guids" => undefined,
                "thumbs:up:user_guids" => undefined,
              },
            },
          }
        }
        navigation={
          Object {
            "push": [MockFunction],
            "state": Object {
              "params": Object {
                "guid": "1",
              },
              "routeName": "some",
            },
          }
        }
        newsfeed={
          NewsfeedStore {
            "loadingBoost": true,
            "service": NewsfeedService {
              "_getFeed": [MockFunction],
              "getBoosts": [MockFunction],
              "getFeed": [MockFunction],
              "getFeedChannel": [MockFunction],
              "getFeedSuggested": [MockFunction],
            },
            "stores": Object {
              "boostfeed": Object {
                "list": OffsetFeedListStore {
                  "entities": Array [],
                  Symbol(mobx administration): ObservableObjectAdministration$$1 {
                    "defaultEnhancer": [Function],
                    "keysAtom": Atom$$1 {
                      "diffValue": 0,
                      "isBeingObserved": false,
                      "isPendingUnobservation": false,
                      "lastAccessedBy": 0,
                      "lowestObserverState": 2,
                      "name": "OffsetFeedListStore@15.keys",
                      "observers": Set {},
                    },
                    "name": "OffsetFeedListStore@15",
                    "target": [Circular],
                    "values": Map {
                      "entities" => Array [],
                    },
                  },
                },
                "loading": false,
              },
              "subscribed": Object {
                "list": OffsetFeedListStore {
                  "entities": Array [],
                  Symbol(mobx administration): ObservableObjectAdministration$$1 {
                    "defaultEnhancer": [Function],
                    "keysAtom": Atom$$1 {
                      "diffValue": 0,
                      "isBeingObserved": false,
                      "isPendingUnobservation": false,
                      "lastAccessedBy": 0,
                      "lowestObserverState": 2,
                      "name": "OffsetFeedListStore@13.keys",
                      "observers": Set {},
                    },
                    "name": "OffsetFeedListStore@13",
                    "target": [Circular],
                    "values": Map {
                      "entities" => Array [],
                    },
                  },
                },
                "loading": false,
              },
              "suggested": Object {
                "list": OffsetFeedListStore {
                  "entities": Array [],
                  Symbol(mobx administration): ObservableObjectAdministration$$1 {
                    "defaultEnhancer": [Function],
                    "keysAtom": Atom$$1 {
                      "diffValue": 0,
                      "isBeingObserved": false,
                      "isPendingUnobservation": false,
                      "lastAccessedBy": 0,
                      "lowestObserverState": 2,
                      "name": "OffsetFeedListStore@14.keys",
                      "observers": Set {},
                    },
                    "name": "OffsetFeedListStore@14",
                    "target": [Circular],
                    "values": Map {
                      "entities" => Array [],
                    },
                  },
                },
                "loading": false,
              },
            },
            "viewed": Array [],
          }
        }
      />
    }
    navigation={
      Object {
        "push": [MockFunction],
        "state": Object {
          "params": Object {
            "guid": "1",
          },
          "routeName": "some",
        },
      }
    }
    onInputFocus={[Function]}
    store={
      Object {
        "attachment": Object {},
        "comments": Array [
          Object {
            "attachment_guid": false,
            "blurb": false,
            "container_guid": "guid0",
            "custom_data": false,
            "custom_type": false,
            "description": "Congratulations! ",
            "edited": "",
            "guid": "guid0",
            "mature": false,
            "ownerObj": Object {
              "guid": "824853017709780997",
              "subtype": false,
              "time_created": "1522036284",
              "type": "user",
            },
            "owner_guid": "824853017709780997",
            "parent_guid": "838106762591510528",
            "perma_url": false,
            "thumbnail_src": false,
            "type": "comment",
          },
          Object {
            "attachment_guid": false,
            "blurb": false,
            "container_guid": "guid1",
            "custom_data": false,
            "custom_type": false,
            "description": "Congratulations! ",
            "edited": "",
            "guid": "guid1",
            "mature": false,
            "ownerObj": Object {
              "guid": "824853017709780997",
              "subtype": false,
              "time_created": "1522036284",
              "type": "user",
            },
            "owner_guid": "824853017709780997",
            "parent_guid": "838106762591510528",
            "perma_url": false,
            "thumbnail_src": false,
            "type": "comment",
          },
          Object {
            "attachment_guid": false,
            "blurb": false,
            "container_guid": "guid2",
            "custom_data": false,
            "custom_type": false,
            "description": "Congratulations! ",
            "edited": "",
            "guid": "guid2",
            "mature": false,
            "ownerObj": Object {
              "guid": "824853017709780997",
              "subtype": false,
              "time_created": "1522036284",
              "type": "user",
            },
            "owner_guid": "824853017709780997",
            "parent_guid": "838106762591510528",
            "perma_url": false,
            "thumbnail_src": false,
            "type": "comment",
          },
          Object {
            "attachment_guid": false,
            "blurb": false,
            "container_guid": "guid3",
            "custom_data": false,
            "custom_type": false,
            "description": "Congratulations! ",
            "edited": "",
            "guid": "guid3",
            "mature": false,
            "ownerObj": Object {
              "guid": "824853017709780997",
              "subtype": false,
              "time_created": "1522036284",
              "type": "user",
            },
            "owner_guid": "824853017709780997",
            "parent_guid": "838106762591510528",
            "perma_url": false,
            "thumbnail_src": false,
            "type": "comment",
          },
          Object {
            "attachment_guid": false,
            "blurb": false,
            "container_guid": "guid4",
            "custom_data": false,
            "custom_type": false,
            "description": "Congratulations! ",
            "edited": "",
            "guid": "guid4",
            "mature": false,
            "ownerObj": Object {
              "guid": "824853017709780997",
              "subtype": false,
              "time_created": "1522036284",
              "type": "user",
            },
            "owner_guid": "824853017709780997",
            "parent_guid": "838106762591510528",
            "perma_url": false,
            "thumbnail_src": false,
            "type": "comment",
          },
        ],
        "embed": RichEmbedStore {
          "_richEmbedFetchTimer": null,
          "clearRichEmbedAction": [Function],
          "richEmbedUrl": "",
          "setRichEmbedPromise": null,
        },
        "loadComments": [Function],
        "loadNext": "aaaaaa",
        "loadPrevious": "aaaaaa",
        "post": [Function],
        "setText": [Function],
      }
    }
  />
</ForwardRef(View)>
`;
